src/main/java/io/rpg/controller/Controller.java

```diff
@@ -168,7 +173,17 @@ private void onAction(BattleAction action) {
     }
     popupController.openTextPopup(result.getMessage(), getWindowCenterX(), getWindowCenterY());
   }
-  
+
+  private void onAction(CollectAction action) {
+    popupController.openTextImagePopup("Picked up an item!", new Image("file:assets/key.png"), getWindowCenterX(), getWindowCenterY());
+//     objectView.setVisible(false);
+    playerController.getPlayer().getInventory().add(action.getOwner());
+    String tag=currentModel.getTag();
+    LocationView currentLocationView=tagToLocationViewMap.get(tag);
+    currentLocationView.removeViewBoundToObject(action.getOwner());
+    currentModel.removeGameObject(action.getOwner());
```

Comment:
```suggestion
    playerController.getPlayer().getInventory().add(action.getOwner());
    String tag = currentModel.getTag();
    LocationView currentLocationView = tagToLocationViewMap.get(tag);
    currentLocationView.removeViewBoundToObject(action.getOwner());
    currentModel.removeGameObject(action.getOwner());
```

+ introduce a variable for the owner instead of calling the getOwner method three times.